app/models/financialDetails/DocumentDetail.scala

```diff
@@ -202,6 +204,10 @@ case class DocumentDetail(taxYear: Int,
     }
   }
 
+  def toTaxYear: TaxYear = {
```

Comment:
not sure if we need this method, seem to be duplicate of makeTaxYearWithEndYear

Reply:
let me check if makeTaxYearWithEndYear works with map

Reply:
It doesn't seem to work with map in the same way (.map(_.taxYear. makeTaxYearWithEndYear). Doing it like this makes it quite concise too I think

Reply:
I would say if we can keep all functions returning same time under relevant singleton ? + if we can avoid creating new function by using existing one, why we need to create something new?